x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryBuilderTests.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.query;
+
+import org.apache.lucene.sandbox.search.CombinedFieldQuery;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.query.CombinedFieldsQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.AbstractQueryTestCase;
+import org.elasticsearch.xpack.relevancesearch.RelevanceSearchPlugin;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RelevanceMatchQueryBuilderTests extends AbstractQueryTestCase<RelevanceMatchQueryBuilder> {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> getPlugins() {
+        List<Class<? extends Plugin>> classpathPlugins = new ArrayList<>();
+        classpathPlugins.add(RelevanceSearchPlugin.class);
+        return classpathPlugins;
+    }
+
+    public void testQueryParsing() throws IOException {
+        String json = """
+            {
+              "relevance_match" : {
+                "query" : "quick brown fox"
+              }
+            }""";
+        RelevanceMatchQueryBuilder parsed = (RelevanceMatchQueryBuilder) parseQuery(json);
+
+        assertEquals(json, "quick brown fox", parsed.getQuery());
+    }
+
+    public void testOptionalParsing() throws IOException {
```

Comment:
Would it make sense to add tests for non-happy path cases where we have invalid data in the query (e.g. a null settings ID?)

Reply:
I'll add them as part of the new parsing exceptions for curations conditions 👍